### PR TITLE
Move tidy check to mingw-check

### LINKED ISSUE #69613: CI: don't block further CI results on tidy/fmt
[Some irlo context.](https://internals.rust-lang.org/t/forced-rustfmt-is-a-roadblock-to-contributing/11913?u=cad97)

Currently, if a PR has even trivial tidy/fmt errors, the tidy/fmt error early-exits the PR smoke test CI and reports just the tidy/fmt error.

This can be problematic because of the long test cycle; it would reduce the number of required pushes/CI runs if CI could report both tidy/fmt nits alongside the test harness state.

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes #69613

### CODE REVIEW

File: src/ci/docker/mingw-check/Dockerfile [hunk 1]
```diff
@@ -25,4 +25,5 @@ ENV RUN_CHECK_WITH_PARALLEL_QUERIES 1
 ENV SCRIPT python2.7 ../x.py check --target=i686-pc-windows-gnu --host=i686-pc-windows-gnu && \
```
Comment (lzutao) [inline src/ci/docker/mingw-check/Dockerfile]: I think tidy should be run ahead of check build.

File: src/ci/docker/x86_64-gnu-llvm-7/Dockerfile [hunk 2]
```diff
@@ -28,7 +28,7 @@ ENV RUST_CONFIGURE_ARGS \
       --enable-llvm-link-shared \
       --set rust.thin-lto-import-instr-limit=10
 
-ENV SCRIPT python2.7 ../x.py test src/tools/tidy && python2.7 ../x.py test
+ENV SCRIPT python2.7 ../x.py test
```
Comment (Mark-Simulacrum) [inline src/ci/docker/x86_64-gnu-llvm-7/Dockerfile]: This won't work. Tidy is already the first thing in the bootstrap list. We'll instead want to do something like this. (We can let the PR builder confirm that I'm doing the right thing).

```
../x.py test --exclude src/tools/tidy && ../x.py test src/tools/tidy
```

**Comment by Mark-Simulacrum** [comment]
I think once the fixes are made this should be able to land. Thanks for the PR!

File: src/ci/docker/x86_64-gnu-llvm-7/Dockerfile (hunk 2 above)
Comment (JohnTitor) [reply src/ci/docker/x86_64-gnu-llvm-7/Dockerfile]: Ah, I didn't know that, addressed.

**Comment by Mark-Simulacrum** [comment]
@bors r+

Thanks!

**Comment by petrochenkov** [comment]
Does this really address https://github.com/rust-lang/rust/issues/69613?

The point of https://github.com/rust-lang/rust/issues/69613 as I understand it is reporting *both* tidy errors and other errors after a single push to GitHub.

Here it still requires two pushes to see e.g. two errors (one substantial and one in formatting), except that now tidy runs last, which is awful because you have to wait 2 hours to see the errors (run, see and fix substantial, run, see tidy) instead of 1.1 hours (quick run, see and fix tidy, run, see substantial).

**Comment by Mark-Simulacrum** [comment]
You will see tidy errors within 10-15 minutes on the mingw-check builder... except the change to the mingw-check builder was lost for some reason? @bors r-

**Comment by JohnTitor** [comment]
@Mark-Simulacrum Ah, I misunderstood your thought, revived the check.

**Comment by Mark-Simulacrum** [comment]
I'm not sure if x.py check first strikes quite the right balance, but we can adjust ordering later too. I think the check builder latency is really low, so I'm not too worried. 

@bors r+

**Pull request merged**